### PR TITLE
ENH: add microsecond format support in read_raw_snirf

### LINKED ISSUE #11788: SNIRF reader can't handle file with MeasurementTime containing milliseconds
### Description of the problem

Can't read .snirf file with `/nirs(i)/metaDataTags/MeasurementTime` in the format `hh:mm:ss.sss`

### Steps to reproduce

```Python
import mne
raw_intensity = mne.io.read_raw_snirf(file, verbose=True).load_data()
```

### Link to data

https://file.io/qgHnp94ioNl7

### Expected results

File should be read without issues since milliseconds should be fine according to the snirf spec (file also passes snirf validator):

image

### Actual results

image

### Additional information

Platform             Windows-10-10.0.19044-SP0
Python               3.11.4 | packaged by conda-forge | (main, Jun 10 2023, 17:59:51) [MSC v.1935 64 bit (AMD64)]
Executable           C:\Users\kdahlslatt\Anaconda3\envs\mne\python.exe
CPU                  Intel64 Family 6 Model 158 Stepping 10, GenuineIntel (12 cores)
Memory               15.7 GB

Core
├☑ mne               1.4.2
├☑ numpy             1.24.3 (MKL 2022.1-Product with 6 threads)
├☑ scipy             1.10.1
├☑ matplotlib        3.7.1 (backend=module://matplotlib_inline.backend_inline)
├☑ pooch             1.7.0
└☑ jinja2            3.1.2

Numerical (optional)
├☑ sklearn           1.2.2
├☑ numba             0.57.0
├☑ nibabel           5.1.0
├☑ nilearn           0.10.1
├☑ dipy              1.7.0
├☑ openmeeg          2.5.6
├☑ pandas            2.0.2
└☐ unavailable       cupy

Visualization (optional)
├☑ pyvista           0.39.1 (OpenGL 4.5.0 - Build 30.0.101.1404 via Intel(R) UHD Graphics 630)
├☑ pyvistaqt         0.0.0
├☑ ipyvtklink        0.2.2
├☑ vtk               9.2.6
├☑ qtpy              2.3.1 (PyQt5=5.15.6)
├☑ pyqtgraph         0.13.3
├☑ mne-qt-browser    0.5.1
└☐ unavailable       ipympl

Ecosystem (optional)
├☑ mne-nirs          0.5.0
└☐ unavailable       mne-bids, mne-features, mne-connectivity, mne-icalabel, mne-bids-pipeline

### PR DESCRIPTION
**Pull request by Xiezhibin**
#### Reference issue
Example: Fixes #11788

#### What does this implement/fix?
This Pull Request implements the parsing of the MeasurementTime data in four different formats:

HH:MM:SS.ffffff (if microsecond is not 0)
HH:MM:SS (if microsecond is 0)
HH:MM:SS.ffffff+HH:MM[:SS[.ffffff]] (if utcoffset() is not None)
HH:MM:SS+HH:MM[:SS[.ffffff]] (if microsecond is 0 and utcoffset() is not None)

### CODE REVIEW

**Review by agramfort — APPROVED** [review]

**Comment by Xiezhibin** [comment]
I'm thrilled to have the opportunity to contribute to this project, and this is my first time submitting a pull request.

I've noticed that my branch is currently out-of-date with the base branch. I would greatly appreciate your guidance on what steps I should take to address this issue and bring my branch up to date. @agramfort @larsoner

**Comment by wmvanvliet** [comment]
If the original MNE-Python repository is named `upstream` in your git setup, the basic command to bring your PR up to date with the current `main` branch is:

```bash
git merge upstream/main
```

This is preferred to rebasing and force pushing. However, your approach also works.

**Comment by wmvanvliet** [comment]
Bugfix-wise, this looks good. Usually, whenever we fix a bug, we try to add a unit test for it (that fails on `main` and succeeds with the new patch). However, this is such a small change I don't think it warrants a test.

You've added your name to `doc/changes/names.inc`, but to cement your place as new contributor further, please edit `doc/changes/latest.inc` as well and add a line under the bugs fixed. This way, this PR shows up in the release notes.

**Comment by Xiezhibin** [comment]
> Bugfix-wise, this looks good. Usually, whenever we fix a bug, we try to add a unit test for it (that fails on `main` and succeeds with the new patch). However, this is such a small change I don't think it warrants a test.
> 
> You've added your name to `doc/changes/names.inc`, but to cement your place as new contributor further, please edit `doc/changes/latest.inc` as well and add a line under the bugs fixed. This way, this PR shows up in the release notes.

Thank you for your insightful response. @wmvanvliet 

- Bugfix-wise: Indeed, for some smaller changes, it may not be necessary to write extensive test cases.  However, I am eager to gain practical experience in testing MNE-Python function modifications.
- merge version: [I have been diligently studying the documentation](https://mne.tools/stable/install/contributing.html#setting-up-your-local-development-environment). it is not recommended to directly use the main branch for submissions. But it recorded an another GitHub Pull Request number,  this is not necessary and i canceled the commit.
- I have already submitted a short description of the changes as requested.

**Comment by Xiezhibin** [comment]
> thanks for the fix! See below for a couple of comments. FYI the advice to not open PRs from your `main` branch is advice to make _your_ life easier, it doesn't make much difference on our side. **After this PR is merged**, here's how to get back to having a `main` branch that matches ours:
> 
> 1. `git checkout -b temp` (switch to temporary branch)
> 2. `git branch -D main`  (delete local `main`)
> 3. `git fetch upstream`  (get the most current upstream verison of `main`)
> 4. `git checkout main upstream/main` (create new local branch `main` based off the current state of `upstream/main`)
> 5. `git branch -D temp`  (delete the temporary branch)

Certainly!  it can get back to the  origin main branch, The fourth revised as follows.

`git checkout -b main upstream/main`

File: doc/changes/names.inc [hunk 1]
```diff
@@ -565,3 +565,5 @@
 .. _Zhi Zhang: https://github.com/tczhangzhi/
 
 .. _Zvi Baratz: https://github.com/ZviBaratz
+
+.. _Daniel Tse: https://github.com/Xiezhibin
```
[the remark this answers is not in the document]
Comment (Xiezhibin) [reply doc/changes/names.inc]: The name has been successfully added to the appropriate spot , ensuring it maintains alphabetical order.

**Review by Xiezhibin — COMMENTED** [review]
After merging, it seems that the code did not pass the tests.

**Pull request merged**